### PR TITLE
[ISSUE #12219] 修复nacos-client中config的ServerListManager解析endpoint失败问题

### LINKED ISSUE #12219: nacos-client中设置endpoint为占位符并且设置isUseEndpointParsingRule为true后运行报错
分支：develop

config中的ServerListManager解析会报错，naming中的ServerListManager正常解析：

image

### PR DESCRIPTION
**Pull request by misakacoder**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

fix #12219 

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: client/src/main/java/com/alibaba/nacos/client/config/impl/ServerListManager.java [hunk 1]
```diff
@@ -316,8 +316,9 @@ private void initEndpoint(final NacosClientProperties properties) {
             String endpointUrl = ParamUtil.parsingEndpointRule(endpointTmp);
             if (StringUtils.isNotBlank(endpointUrl)) {
                 this.serverAddrsStr = "";
+                this.endpoint = endpointUrl;
+                return;
             }
-            this.endpoint = endpointUrl;
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/config/impl/ServerListManager.java:320]: 是不是这行改成`endpointTmp=endpointUrl` 即可

Comment (misakacoder) [reply client/src/main/java/com/alibaba/nacos/client/config/impl/ServerListManager.java:320]: 这样改也可以

Comment (KomachiSion) [reply client/src/main/java/com/alibaba/nacos/client/config/impl/ServerListManager.java:320]: 那麻烦改一下

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Comment by misakacoder** [comment]
@KomachiSion 改了

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**